scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -1519,10 +1530,17 @@
                      ;; TODO: Rescind the offer in fenzo
                      )
     (framework-message [this driver executor-id slave-id message]
-                       ;; the executor-id is the same as the task-id (see cook.mesos.task/task-info->mesos-message)
-                       (async-in-order-processing (:value executor-id)
-                                                  #(handle-framework-message conn handle-progress-message message))
-                       (heartbeat/notify-heartbeat heartbeat-ch executor-id slave-id message))
+                       (try
+                         (let [{:strs [task-id] :as parsed-message} (json/read-str (String. ^bytes message "UTF-8"))
+                               executor (task-id->executor (d/db conn) task-id)]
+                           (when (some #(not (contains? #{"task-id" "timestamp"} %)) (keys parsed-message))
```

Comment:
Can we be more explicit about what we're handling here?  Perhaps we need a `:message-type` field.

Reply:
The executor is free to merge multiple values into a message. We are optimizing for filtering out heartbeat messages that only provide the timestamp. Would you be happy with a comment explicitly mentioning that we are filtering out messages with only heartbeat values?

Reply:
Also, with the `message-type` field we will then need all executors (i.e. custom and mesoso along with cook executor) to send this field.